src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -93,10 +104,53 @@ public static int Run(string[] args)
 
                 msbuildArgs.AddRange(app.RemainingArguments);
 
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                return 0;
             });
 
-            return app.Execute(args);
+            int exitCode = app.Execute(args);
+            if (msbuildArgs == null)
```

Comment:
Why not check the exit code of app.Execute instead of checking this variable. Depending on this variable seems extremely fragile to me. All that you really are trying to do is answer the question: did parsing succeed?